### PR TITLE
Makes borg buckling conflict less with inventory searching

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

This PR _finally_ splits up the ways that borgs are supposed to initiate inventory checks of a mobs and initiate attempts to buckle mobs to themselves (formerly, they would both happen upon click+dragging a mob onto yourself as a borg). Now a borg will only try to buckle a mob to themself if they ARE NOT on harm intent when they finish click+dragging said mob onto themself, and will only try to check a mob's inventory if they ARE on harm intent when they finish click+dragging said mob onto themself. An outsider can still buckle someone (even themself) to a borg who's on harm intent; the borg just can't do it themself.

## Why It's Good For The Game

This issue is incredibly annoying to deal with as a borg (especially if you're in an area or situation where *spinning someone off of you could be risky), and I'm really surprised that it's taken this long for this control conflict to be fixed.

## Changelog
:cl: ATHATH
tweak: If a cyborg click+drags someone onto themself, they will only try to buckle that person to themself if they (the cyborg) ARE NOT on harm intent. Attempts made by non-cyborgs to buckle themselves to cyborgs by click+dragging themselves onto said cyborgs should be unaffected by this change, even if those cyborgs are on harm intent.
tweak: If a cyborg click+drags someone onto themself, they will only attempt to check that person's inventory if they (the cyborg) ARE on harm intent.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
you suck at wording but i'm assuming this change makes it so trying to strip people doesn't buckle them into you, so thanks 👍

**Comment by mrhugo13** [comment]
speedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmergespeedmerge

**Comment by Ratvar** [comment]
Now make it impossible to buckle someone onto yourself if their hand is inactive to fix the entire issue called flipstun

**Comment by Fhaxaris** [comment]
Now make clicking yourself with no module unbuckle them.

**Comment by kriskog** [comment]
> 
> 
> Now make it impossible to buckle someone onto yourself if their hand is inactive to fix the entire issue called flipstun

That was done so borgs could pick up unconscious/dead people after the drag slowdown change.

**Comment by ATH1909** [comment]
> Now make clicking yourself with no module unbuckle them.

I might look into that for a second PR, actually.

**Comment by zxaber** [comment]
As an alternative, you might consider help/harm intent. Either one is a single button press to toggle, but intent is already used elsewhere to determine between conflicting actions.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -1133,7 +1133,7 @@
 
 /mob/living/silicon/robot/MouseDrop_T(mob/living/M, mob/living/user)
 	. = ..()
-	if(!(M in buckled_mobs) && isliving(M))
+	if(!(M in buckled_mobs) && isliving(M) && ((user!=src)||!(src.get_active_held_item())))
```
Comment (zxaber) [inline code/modules/mob/living/silicon/robot/robot.dm]: If you stick a return inside the if block right after `buckle_mob(M)`, and then move the `. = .. ()` to after the if block, you probably won't need the changes you have in mob.dm.

**Comment by ATH1909** [comment]
> As an alternative, you might consider help/harm intent. Either one is a single button press to toggle, but intent is already used elsewhere to determine between conflicting actions.

Hm, that might indeed be a better solution. Which should be which, though? Should harm intent be for buckling and help intent for searching, or the other way around?

**Comment by zxaber** [comment]
I would say help for buckle, harm for strip, if only because stripping is _usually_ not time-sensitive while buckling a human might be followed immediately by running from something like a plasma fire or xenos.

**Comment by optimumtact** [comment]
can you giv eyoru PR a title

**Comment by ATH1909** [comment]
> I would say help for buckle, harm for strip, if only because stripping is _usually_ not time-sensitive while buckling a human might be followed immediately by running from something like a plasma fire or xenos.

Alright, this PR now makes the buckle vs. search check check for what intent you're on instead of whether or not you're holding an item.

**Pull request merged**